### ISSUE TITLE
SDL2 linker issue building 

### ISSUE THREAD

**Issue by ColeW-Picaro** [eduard-permyakov/permafrost-engine#26]
Following the build instructions for windows, and `make pf PLAT=WINDOWS` fails when building on windows. I get this error:
![image](https://user-images.githubusercontent.com/16948412/118052498-d6f3f580-b350-11eb-90ce-913e8293f8d4.png)

Other sources say this normally comes in the `python-dev` package, but that's not available for windows.

**Comment by eduard-permyakov** [comment]
Did you do a `make deps PLAT=WINDOWS` to build CPython before this step? This file is generated during the Python build process and has some config-specific and platform-specific stuff.

**Comment by ColeW-Picaro** [comment]
I did, but it looks like it errored building SDL2 without me noticing. Getting a lot of linking errors when running `make deps`:
![2021-05-13-145632_1867x651_scrot](https://user-images.githubusercontent.com/16948412/118174075-5e4a7300-b3fc-11eb-8355-b4255a401fbc.png)

**Comment by eduard-permyakov** [comment]
Okay, so there's no real "secret sauce" to the SDL build. The top-level Makefile essentially just calls `make` in the SDL2 directory, which builds the unmodified source in a standard manner.

As a first experiment, try to clone the default SDL (ex. from here: https://github.com/libsdl-org/SDL) and do `./configure` followed by `make`. If it doesn't work, then you have a problem on your side with your toolchain or something.

It the default SDL build works, then just double check you're not mixing `make deps` and `make deps PLAT=WINDOWS` (I noticed you mismatched them in your 2 comments). Also make sure you're either building from a Linux box (recommended) or from Windows using MSYS, and not from something like macOS. If those checks pass, try `make clean_deps` and re-build. Let me know how it goes.

P.S. As a workaround you can also just copy `SDL2.dll` to the `./lib` directory.

**Comment by ColeW-Picaro** [comment]
I mismatched them in my comments because I've tried on both Windows and Linux. I'm trying on Linux right now.

I built SDL and it looks like it linked properly. I ran `make clean_deps` and `make deps` again and got the same errors. Here's a screenshot with where they differ. Running `make deps` on the left, running `make` for SDL on the right.
![2021-05-14-184100_1879x1008_scrot](https://user-images.githubusercontent.com/16948412/118338369-fcfdcf00-b4e3-11eb-888a-ae5133945df6.png)

Looks like SDL built. The warning at the bottom is from a test, so I don't think the build failed.
